colfetcher: increase input batch size limit when using the Streamer

yuzefovich opened the pull request:
This commit bumps up the input batch size limit in the vectorized index
join from 4MiB to 8MiB when the streamer API is used. The old number was
copy-pasted from the row-by-row join reader, and the new number is
derived based on running TPCH queries 4, 5, 6, 10, 12, 14, 15, 16 using
`tpchvec/bench`. I did a quick run of the same queries with the
row-by-row engine used by default, and it seemed like the default of
4MiB there is reasonable, so I didn't change that one.

With the non-streamer code path we didn't want to go higher than 4MiB
since it showed diminishing returns while exposing the cluster to higher
instability (since the non-streamer code path doesn't use the memory
limits in the KV layer). The streamer does use memory limits for
BatchRequests, so it should be safe to increase this limit.

Additionally, this commit makes it so that 1/8th (rather than 1/4th) of
the workmem limit is reserved for the output batch of the cFetcher. With
the default value of 64MiB this would translate to 8MiB which is large
enough.

Addresses: #82159.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82305)
<!-- Reviewable:end -->

yuzefovich committed:
colfetcher: increase input batch size limit when using the Streamer

This commit bumps up the input batch size limit in the vectorized index
join from 4MiB to 8MiB when the streamer API is used. The old number was
copy-pasted from the row-by-row join reader, and the new number is
derived based on running TPCH queries 4, 5, 6, 10, 12, 14, 15, 16 using
`tpchvec/bench`. I did a quick run of the same queries with the
row-by-row engine used by default, and it seemed like the default of
4MiB there is reasonable, so I didn't change that one.

With the non-streamer code path we didn't want to go higher than 4MiB
since it showed diminishing returns while exposing the cluster to higher
instability (since the non-streamer code path doesn't use the memory
limits in the KV layer). The streamer does use memory limits for
BatchRequests, so it should be safe to increase this limit.

Additionally, this commit makes it so that 1/8th (rather than 1/4th) of
the workmem limit is reserved for the output batch of the cFetcher. With
the default value of 64MiB this would translate to 8MiB which is large
enough.

Release note: None

rharding6373 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks for doing the analysis!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/82305#-:-N3edNaK4zEQrjF00BMu:bo32wi6)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @michae2 and @yuzefovich)

___
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/82305#-N3ecNoUDLJb5be2HeZH):*
What batch size limits did you try?


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/82305#-:-N3eqKgqARetPZ7SUQ0G:b-oduykh)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @michae2 and @rharding6373)

___
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/82305#-N3ecNoUDLJb5be2HeZH):*
<details><summary><i>Previously, rharding6373 (Rachael Harding) wrote…</i></summary><blockquote>

What batch size limits did you try?
</blockquote></details>

I think I used 1MiB, 4MiB, 8MiB, 12MiB, 16MiB.


<!-- Sent from Reviewable.io -->

yuzefovich commented:
Needs minor adjustments.

bors r-

yuzefovich commented:
bors r+

yuzefovich commented:
bors r+

yuzefovich commented:
Unrelated flake.

bors r+
